### ISSUE TITLE
[request]: namespace available for "Spot Allocation Strategy"

### ISSUE THREAD

**Issue by scanianick** [aws/elastic-beanstalk-roadmap#36]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Tell us about your request**
What do you want us to build?
Namespace available for "Spot Allocation Strategy". 

**Is this request specific to an Elastic Beanstalk [platform](https://docs.aws.amazon.com/elasticbeanstalk/latest/platforms/platforms-supported.html)?**
If so, which one(s)?
No

**Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?**
What outcome are you trying to achieve, ultimately, and why is it hard/impossible to do right now? What is the impact of not having this problem solved? The more details you can provide, the better we'll be able to understand and solve the problem.

It would be great to have support for "Spot Allocation Strategy" for the ASG namespace.

**Are you currently working around this issue?**
How are you currently solving this problem?
Just selecting one instance type for spot instances instead of higher ends as well and fail over to ondemand instead.

**Additional context**
Anything else we should know?

**Attachments**
If you think you might have additional information that you'd like to include via an attachment, please do - we'll take a look. (Remember to remove any personally-identifiable information.)

**Comment by stefansundin** [comment]
The current way to work around this is to create an ebextension (e.g. `.ebextensions/resources.config`) with the following contents:

```yaml
Resources:
  AWSEBAutoScalingGroup:
    Type: "AWS::AutoScaling::AutoScalingGroup"
    Properties:
      MixedInstancesPolicy:
        InstancesDistribution:
          SpotAllocationStrategy: "lowest-price"
```

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-autoscaling-autoscalinggroup-instancesdistribution.html#cfn-autoscaling-autoscalinggroup-instancesdistribution-spotallocationstrategy

If you try to include this code for an environment that is not using a launch template, then the deployment will fail. So just keep that in mind.

**Comment by Sekhar-Kutikuppala** [comment]
@scanianick Elastic Beanstalk uses [the new capacity-optimized allocation strategy](https://aws.amazon.com/about-aws/whats-new/2019/08/new-capacity-optimized-allocation-strategy-for-provisioning-amazon-ec2-spot-instances/) for provisioning Spot instances. I'm curious if you prefer lowest-price to capacity-optimized and any particular requirements to change the allocation strategy?

**Comment by mksenzov** [comment]
@Sekhar-Kutikuppala yes, there are cases when "lowest-price" is the best strategy, e.g. cases when smallest reasonable instance type is fine, anything below it is too small and you add bigger instances only to be able to get your spot requests serve in time. It happens when you mostly rely on the smallest instance in the given family (g4dn.large could be the one).

**Comment by azpaulp** [comment]
We are happy to report that this issue is resolved. Please refer to [the What's New post](https://aws.amazon.com/about-aws/whats-new/2025/01/aws-elastic-beanstalk-spot-allocation-strategies/#:~:text=AWS%20Elastic%20Beanstalk%20is%20expanding,existing%20default%20capacity%2Doptimized%20strategy.) for more information.
